felix/dataplane/linux/endpoint_mgr.go

```diff
@@ -669,12 +669,14 @@ func (m *endpointManager) resolveWorkloadEndpoints() {
 	}
 
 	removeActiveWorkload := func(logCxt *log.Entry, oldWorkload *proto.WorkloadEndpoint, id types.WorkloadEndpointID) {
-		// QoS state must be removed before the workload itself is removed
-		if oldWorkload != nil {
-			logCxt.Info("deleting QoS bandwidth state if present")
-			err := m.maybeUpdateQoSBandwidth(oldWorkload, nil)
-			if err != nil {
-				logCxt.WithField("workload", oldWorkload).Infof("error deleting QoS bandwidth state, workload may have been already removed: %v", err)
+		if !m.bpfEnabled {
+			// QoS state must be removed before the workload itself is removed
+			if oldWorkload != nil {
+				logCxt.Info("deleting QoS bandwidth state if present")
+				err := m.maybeUpdateQoSBandwidth(oldWorkload, nil)
+				if err != nil {
+					logCxt.WithField("workload", oldWorkload).Infof("error deleting QoS bandwidth state, workload may have been already removed: %v", err)
```

Comment:
```suggestion
					logCxt.WirthError(err).WithField("workload", oldWorkload).Info("Error deleting QoS bandwidth state, workload may have been already removed.")
```
I prefer using `WithError` if we're using `WithField`

Will this log fire for all endpoints that never had QoS?  Could be spammy in that case; would be good to make it a debug log if it's an "expected" error (such as interface/qdisc not found).